nbodykit/dataset.py

```diff
@@ -0,0 +1,849 @@
+                
+        return self.__finalize__(self.data[key], self.mask[key], indices)
+        
+    def _get_index(self, dim, val):
+        """
+        Internal function to compute the bin index of the nearest 
+        coordinate value to the input value
+        """
+        index = self.coords[dim]
+        if self.force_index_match:
+            i = (numpy.abs(index-val)).argmin()
+        else:
+            try:
+                i = list(index).index(val)
+            except Exception as e:
+                args = (dim, str(e))
+                msg = "error converting '%s' index; try setting `force_index_match=True`: %s"
+                raise IndexError(msg %args)
+                
+        return i
+        
+    #--------------------------------------------------------------------------
+    # user-called functions
+    #--------------------------------------------------------------------------
+    def sel(self, **kwargs):
+        """
+        Label-based indexing by coordinate name. Indexing should be supplied
+        using the dimension name as the key and the desired coordinate values as
+        the keyword value.
+        
+        Returns
+        -------
+        sliced : DataSet
+            a new ``DataSet`` holding the sliced data and coordinate grid
+        
+        
+        Examples
+        --------
+        >>> pkmu
+        <DataSet: dims: (k_cen: 200, mu_cen: 5), variables: ('mu', 'k', 'power')>
+        >>> sliced_1 = pkmu.sel(k_cen=0.4)
+        >>> sliced_1
+        <DataSet: dims: (k_cen: 1, mu_cen: 5), variables: ('mu', 'k', 'power')>
+        >>> sliced_2 = pkmu.sel(k_cen=slice(0.1, 0.4), mu_cen=0.5)
+        >>> sliced_2
+        <DataSet: dims: (k_cen: 30, mu_cen: 1), variables: ('mu', 'k', 'power')>
+        """
+        indices = [range(0, self.shape[i]) for i in range(self.ndim)]
+        for dim in kwargs:
+            key = kwargs[dim]
+            i = self.dims.index(dim)
+            
+            if isinstance(key, list):
+                indices[i] = [self._get_index(dim, k) for k in key]
+            elif isinstance(key, slice):
+                new_slice = []
+                for name in ['start', 'stop']:
+                    new_slice.append(self._get_index(dim, getattr(key, name)))
+                indices[i] = range(*slice(*new_slice).indices(self.shape[i]))
+            else:
+                indices[i] = [self._get_index(dim, key)]
+        
+        # check for empty slices
+        for i, idx in enumerate(indices):
+            if not len(idx):
+                raise KeyError("trying to use empty slice for dimension '%s'" %self.dims[i])
+        
+        data = self.data.copy()
+        mask = self.mask.copy()
+        for i, idx in enumerate(indices):
+            data = numpy.take(data, idx, axis=i)
+            mask = numpy.take(mask, idx, axis=i)
+        
+        return self.__finalize__(data, mask, indices)
+        
+    def to_pickle(self, filename):
```

Comment:
I'm just pickling the components needed to reconstruct the class -- mainly to avoid issues with pickling while the code is still in flux, i.e., the class name changes. I've been burned by that before, but I'm open to just doing the standard pickling